### ISSUE TITLE
Production mode doesn't work in IT tests with NPM

### ISSUE THREAD

**Issue by denis-anisimov** [vaadin/flow#5727]
Use this PR to check:
#5716

Step to reproduce:

- Build Flow.
- go into `flow-tests/test-root-context`.
- run `mvn  -f pom-npm.xml install -Pproduction -DskipTests`
- run `mvn  -f pom-npm.xml jetty:run -Pproduction`
Open e.g. http://localhost:8888/view/com.vaadin.flow.uitest.ui.template.EmptyListsView.
The result is: 404 for `index.js` and `webcomponentsjs/webcomponents-loader.js`.
Also there is an error: 
```
There was an exception while trying to navigate to 'com.vaadin.flow.uitest.ui.template.EmptyListsView' with the exception message 'Unable to create an instance of 'com.vaadin.flow.uitest.ui.template.EmptyListsView'. The constructor threw an exception.'
```
The expected result: no any 404 for any resource and no error.

This is related to the #5726. May be have similar reasons.
But the workaround with PR #5725 doesn't help here.

**Comment by bogdanudrescu** [comment]
I think this is duplicate of https://github.com/vaadin/flow/issues/5740

**Comment by denis-anisimov** [comment]
Most likely yes: will be fixed along with fix for #5740.
But this is a separate ticket which should be verified after #5740 is fixed.
So please don't close it.

**Comment by denis-anisimov** [comment]
"Flow Snapshots Validation with NPM production" build job is created to run NPM production mode IT tests for `test-root-context` .
It passes now and it should be triggered by a snapshot.
